cni/pkg/zdsapi/zds.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// GRPC package - part of the URL. Service is added.
+// URL: /PACKAGE.SERVICE/METHOD
+package istio.workload.zds;
+
+option go_package="pkg/zdsapi";
+
+// Add a workload to the ztunnel. Ancillary data containing
+// the workload's netns file descriptor will be appended to the end of this message.
+message AddWorkload {
+  string uid = 2;
+}
+
+// Keep workload that we can't find in the fd cache. This can only be sent before SnapshotSent is sent
+// to signal ztunnel to not delete the workload if it has it.
+message KeepWorkload {
+  string uid = 1;
+}
+
+// Delete a workload from the ztunnel. Ztunnel should shutdown the workload's proxy.
+message DelWorkload {
+  string uid = 2;
+}
+
+// Let ztunnel know that a full snapshot was sent. Ztunnel should reconcile its internal state
+// and remove internal entries that were not sent.
+message SnapshotSent {
+}
+
+// Let ztunnel know that no snapshot will be sent, and no reconciliation is needed.
+// will be removed once we wrap up the CNI, here for now to help with testing.
+message NoSnapshotPresent {
```

Comment:
never used?

Reply:
yes, good point; IIRC was used in early iterations (as the ztunnel had snapshot support before the agent); I think we can  remove

Reply:
I'll remove it here, it's also included (but obviously also unused) in the sibling PR.